### DISCUSSION TITLE
How to get ‘baseRoute’ automatically from the dir

### DISCUSSION THREAD

**Discussion by wssnail** [hannoeru/vite-plugin-pages#398] [Q&A]
dirs: [
          { dir: 'src/views/**/pages', baseRoute: '' },
          { dir: 'src/views/abc/pages', baseRoute: 'abc' },
        ],

Theoretically, you can get it from extendRoute, and then modify the path prefix, but is there any other better way

**Comment by reslear** [comment]
duplicate #15

**Comment by reslear** [comment]
so good is implemented out of the box, my solution: 

```ts
import fg from 'fast-glob'

// ... 

dirs: [
	...fg
	  .sync('src/modules/**/pages', {
	    onlyDirectories: true,
	  })
	  .map((dir) => ({
	      dir,
	      baseRoute: dir.split('/').slice(-2)[0],
	    })
	  ),
	]
]
```
